### ISSUE TITLE
camera permission asked and crash at start on iOS with module_arkit_enabled=no

### ISSUE THREAD

**Issue by volzhs** [godotengine/godot#30261]
**Godot version:**

3.2.dev 

**OS/device including version:**

iOS 12.3.1 / iPad 6th gen

**Issue description:**

I don't need arkit so I compiled iOS templates with `module_arkit_enabled=no`
problems are, when run on device
1. it asks camera permission
2. crashes after permission check

**Steps to reproduce:**
1. compile iOS templates with `module_arkit_enabled=no`
2. run on device

**Minimal reproduction project:**

**Comment by volzhs** [comment]
I tested with `module_arkit_enabled=yes` and added `ARKit` framework on xcode.
and still crashes at same point.
```
Godot Engine v3.2.dev.custom_build.726a2b46b - https://godotengine.org
OpenGL ES 3.0 Renderer: Apple A10 GPU
Requesting Camera permissions
[MC] System group container for systemgroup.com.apple.configurationprofiles path is /private/var/containers/Shared/SystemGroup/systemgroup.com.apple.configurationprofiles
[MC] Reading from public effective user settings.
Access to cameras granted!
(lldb) 
```

**Comment by BastiaanOlij** [comment]
I ran into this earlier in the week too, I'm not sure why. It's not related to ARKit but general camera access.

When Godot starts the new camera server will attempt to get access to the list of cameras. For some reason we're getting permission for this while no permission has been given, when you then try and access the cameras to create the listing it crashes. 

Basically when you don't specify a permissions string in the export (ends up in your plist) it shouldn't ask for camera permissions and fail the permissions check. Possibly we need to explicitly check for this condition instead of relying on iOS to handle this.

@akien this looks very much like the issues we were having with the microphone API as well.

**Comment by BastiaanOlij** [comment]
Just in case someone has time to fix this before I can get to it, the code that needs to improve is this:
https://github.com/godotengine/godot/blob/master/platform/iphone/camera_ios.mm#L402

My gut feeling is that we should add a check here to see if the plist entry for the camera permission string (i.e. this one: https://github.com/godotengine/godot/blob/master/misc/dist/ios_xcode/godot_ios/godot_ios-Info.plist#L36 ) is set and only then run the code in this method.

Alternatively, we should check that if you don't give a permission string in the export that it indeed clears/removes this entry. That could possibly also be an issue.

**Comment by volzhs** [comment]
I confirmed that reverting ARKit still asked camera permission on ios device.

**Comment by pip-byte** [comment]
Is there any hot fix for this problem?
At the moment it is **not possible** to release ANY Godot App on the iOS App store.
Godot automatically asks for camera access at startup wich causes the app to crash on certain devices and leads to the app rejection when you try to publish it.

I could reproduce the issue on an iPhone 5S using iOS12 (works fine on iPhone X and iPad Pro and iOS13). The App does not crash when you delete the "Privacy - Camera Usage Description" but Godot still tries to get permission and the App can not be uploaded to the App store.

I am trying to release a game for children and, apart from being unable to publish it, Godot asking for Camera access on startup is quite problematic.

My code does not contain any references to the camera, microphone or arkit. Everything is deselected and I have even tried to exclude them using the entitlement description.

I have tried Godot 3.1, Godot 3.2 Alpha and both Betas. I have also tried to revert to 3.1.2RC1 but it does not recognize my .tscn files after they have been opened in 3.2

I would be grateful for any help on this matter.
